js/blocks/PitchBlocks.js

```diff
@@ -1358,6 +1358,172 @@ function setupPitchBlocks() {
+                }
+            }
+        }
+    }
+
+    class SolfegeSyllableBlock extends ValueBlock {
+        constructor() {
+            //.TRANS: print the solfege e.g. do, re, mi 
+            super("solfegesyllable", _("solfege syllable"));
+            this.setPalette("pitch");
+            this.parameter = true;
+            this.setHelpString([
+                _(
+                    "The Solfege Syllable block return the solfege subject to note currently being played and whether or not moveable do is present"
+                ),
+                "documentation",
+                ""
+            ]);
+            this.beginnerBlock(true);
+        }
+
+        updateParameter(logo, turtle, blk) {
+            return logo.blocks.blockList[blk].value;
+        }
+
+        arg(logo, turtle, blk) {
+            if (
+                logo.inStatusMatrix &&
+                logo.blocks.blockList[logo.blocks.blockList[blk].connections[0]]
+                    .name === "print"
+            ) {
+                logo.statusFields.push([blk, "solfegesyllable"]);
+            } else {
+                if (logo.noteStatus[turtle] != null){
+                    let letterClass = logo.noteStatus[turtle][0][0][0];
+                    if (logo.moveable[turtle] === false) {
+                        return SOLFEGECONVERSIONTABLE[letterClass];
+                    } else {
+                        let scale = _buildScale(logo.keySignature[turtle])[0];
+                        let i = scale.indexOf(letterClass);
+                        return SOLFEGENAMES[i];
+                    }
+                } else {
+                    return "";
+                }
+            }
+        }
+    }
+
+    class PitchCLassBlock extends ValueBlock {
+        constructor() {
+            //.TRANS: print the solfege e.g. do, re, mi 
+            super("pitchclass", _("pitch class"));
+            this.setPalette("pitch");
+            this.parameter = true;
+            this.setHelpString([
+                _(
+                    "The Pitch Class block returns the letter class of note currently being played and also the accidental data"
+                ),
+                "documentation",
+                ""
+            ]);
+            this.beginnerBlock(true);
+        }
+
+        updateParameter(logo, turtle, blk) {
+            return logo.blocks.blockList[blk].value;
+        }
+
+        arg(logo, turtle, blk) {
+            if (
+                logo.inStatusMatrix &&
+                logo.blocks.blockList[logo.blocks.blockList[blk].connections[0]]
+                    .name === "print"
+            ) {
+                logo.statusFields.push([blk, "pitchclass"]);
+            } else {
+                if (logo.noteStatus[turtle] !== null) {
+                    let note = logo.lastPitchPlayed[0][0];
+                    let pitchClass = note[0];
+                    console.log(note);
+                    if (note.indexOf("#") != -1) {
```

Comment:
let us use contains it makes code more readable.

Reply:
Okay. Will address this in the next commit.